jcli/src/jcli_lib/transaction/staging.rs

```diff
@@ -358,10 +375,21 @@ impl Staging {
         }
 
         match &self.extra {
-            None => {
-                self.finalize_payload(&chain::transaction::NoExtra, fee_algorithm, output_policy)
-            }
-            Some(ref c) => match c.clone().into() {
+            None => match &self.evm_transaction {
+                None => self.finalize_payload(
+                    &chain::transaction::NoExtra,
+                    fee_algorithm,
+                    output_policy,
+                ),
+                Some(tx) => {
+                    #[allow(unreachable_code)]
+                    #[allow(unused_variables)]
+                    let tx: chain::evm::EvmTransaction = tx.clone().into();
+                    #[allow(unreachable_code)]
+                    self.finalize_payload(&tx, fee_algorithm, output_policy)
+                }
+            },
```

Comment:
Are these really unreachable/unused now?

Reply:
We have the following situation at this place. When we are compiling with the `feature = "evm"` so it works fine and compiles with any problem, but when we are compiling without  `feature = "evm"` so `chain::evm::EvmTransaction` become a `uninhabited` type, so it is not way how to initialize a value with this type so according to this this code will be unreachable.
```
pub enum EvmTransaction {}
```

Maybe it is better than I will do it in the following way
```
#[cfg(feature(evm)]
{
     let tx: chain::evm::EvmTransaction = tx.clone().into();
     self.finalize_payload(&tx, fee_algorithm, output_policy)
}
#[cfg(feature(evm)]
unreachable!()
```

Reply:
That would be the correct approach IMO. These unreachables would hit us in the back later.